tests/components/pi_hole/test_sensor.py

```diff
@@ -0,0 +1,39 @@
+"""Test pi_hole sensor."""
+
+from homeassistant.components.pi_hole import (
+    sensor as pi_hole,
+    DOMAIN as PIHOLE_DOMAIN,
+    PiHoleData,
+)
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+
+
+added_sensors = []
+
+
+def mock_async_add_entities(sensors, enable):
+    """Mock function."""
+    global added_sensors
+    added_sensors = sensors
+    return
+
+
+async def test_setup_no_monitored_conditions(hass):
+    """Test a successful setup with no monitored_conditions."""
+    hass.data[PIHOLE_DOMAIN] = PiHoleData(None, None)
+
+    config = {CONF_MONITORED_CONDITIONS: []}
+    await pi_hole.async_setup_platform(hass, config, mock_async_add_entities, None)
```

Comment:
Instead of using global in your mock, do something like:

```python
entities = []
mock_async_add_entities = entities.extend
```

Although generally it's better if you just test the states that are added to the state machine. That way you could also verify their data.

Reply:
Please set up the pi_hole component which should set up the sensor platform, let it add an entity, then assert the state of the entity via the core state machine.

Use `async_setup_component` from `homeassistant/setup.py` to set up the component.

If needed, patch time and fire time changed event to control time and move it forward to make the entity poll for new data. See eg:
https://github.com/home-assistant/home-assistant/blob/ab7db5fbd0c7261fbd1b2e7feeceaeea0c1b4f3a/tests/components/manual/test_alarm_control_panel.py#L83-L119